sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -869,6 +1000,9 @@ case class GpuRowNumber() extends GpuAggregateWindowFunction[RowNumberAggregatio
     assert(inputs.isEmpty, inputs)
     Aggregation.rowNumber().onColumn(0)
   }
+
+  override def newFixer(): BatchedRunningWindowFixer =
+    new BatchedRunningWindowBinaryFixer(BinaryOp.ADD, "row_number")
```

Comment:
I assume this is a kind of no-op for row_number, and that the partitioner is looking at all columns.

Reply:
`row_number` only applies within a given partition.  The output is identical to doing a `count(1)`. The main difference is that `row_number` only supports a running window so there are no odd corner cases when you ask for a row number.

```
scala> val df = spark.range((20).toLong).selectExpr("id DIV 10 as a", "id as b")
df: org.apache.spark.sql.DataFrame = [a: bigint, b: bigint]
 
scala> spark.time(df.coalesce(1).selectExpr("a", "b", "row_number() over (partition by a order by b) as row_num", "count(1) over (partition by a order by b rows between UNBOUNDED PRECEDING and CURRENT ROW) as count_row_num" ).show(100)) 
+---+---+-------+-------------+
|  a|  b|row_num|count_row_num|
+---+---+-------+-------------+
|  0|  0|      1|            1|
|  0|  1|      2|            2|
|  0|  2|      3|            3|
|  0|  3|      4|            4|
|  0|  4|      5|            5|
|  0|  5|      6|            6|
|  0|  6|      7|            7|
|  0|  7|      8|            8|
|  0|  8|      9|            9|
|  0|  9|     10|           10|
|  1| 10|      1|            1|
|  1| 11|      2|            2|
|  1| 12|      3|            3|
|  1| 13|      4|            4|
|  1| 14|      5|            5|
|  1| 15|      6|            6|
|  1| 16|      7|            7|
|  1| 17|      8|            8|
|  1| 18|      9|            9|
|  1| 19|     10|           10|
+---+---+-------+-------------+

Time taken: 293 ms


```